internal/exec/engine.go

```diff
@@ -61,14 +63,21 @@ func (e Engine) Run(stageName string) bool {
 		},
 	}
 
+	f, err := e.OEMConfig.NewFetcherFunc()(e.Logger)
```

Comment:
I think this should be outside of `Run()` in `main()` where the rest of the engine struct is assembled